tests/tests_raid_pool_options.yml

```diff
@@ -0,0 +1,72 @@
+---
+- hosts: all
+  become: true
+  vars:
+    storage_safe_mode: false
+    storage_use_partitions: true
+    mount_location1: '/opt/test1'
+    mount_location2: '/opt/test2'
+    mount_location3: '/opt/test3'
+    volume1_size: '2g'
+    volume3_size: '3g'
+    volume2_size: '3g'
+
+  tasks:
+    - include_role:
+        name: storage
+
+    - include_tasks: get_unused_disk.yml
+      vars:
+        max_return: 3
+
+    - name: Create a RAID1 device
+      include_role:
+        name: storage
+      vars:
+        storage_pools:
+          - name: vg1
+            disks: "{{ unused_disks }}"
+            type: lvm
+            raid_level: "raid1"
+            raid_device_count: 2
+            raid_spare_count: 1
+            raid_metadata_version: "1.0"
+            state: present
+            volumes:
+              - name: lv1
+                size: "{{ volume1_size }}"
+                mount_point: "{{ mount_location1 }}"
+              - name: lv2
+                size: "{{ volume2_size }}"
+                mount_point: "{{ mount_location2 }}"
+              - name: lv3
+                size: "{{ volume3_size }}"
+                mount_point: "{{ mount_location3 }}"
+
+    - include_tasks: verify-role-results.yml
```

Comment:
I don't think there is anything to validate that the optional parameters were applied correctly.

Reply:
Fixed: Added new file with raid pool options tests.